client/file-manager/src/rocksdb.rs

```diff
@@ -0,0 +1,1023 @@
+            _marker: Default::default(),
+        })
+    }
+}
+
+impl<T: TrieLayout + 'static + Send + Sync> FileStorage<T> for RocksDbFileStorage<T>
+where
+    HasherOutT<T>: TryFrom<[u8; H_LENGTH]>,
+{
+    type FileDataTrie = RocksDbFileDataTrie<T>;
+
+    fn get_chunk(
+        &self,
+        key: &HasherOutT<T>,
+        chunk_id: &ChunkId,
+    ) -> Result<Chunk, FileStorageError> {
+        let raw_metadata = self
+            .storage
+            .read()
+            .expect("Failed to acquire read lock")
+            .read(*key)
+            .map_err(|_| FileStorageError::FailedToReadStorage)?
+            .expect("Failed to find File Metadata");
+
+        let metadata: FileMetadata = serde_json::from_slice(&raw_metadata)
+            .map_err(|_| FileStorageError::FailedToParseFileMetadata)?;
+        let raw_root = metadata.fingerprint.as_ref();
+        let mut root = convert_raw_bytes_to_hasher_out::<T>(raw_root.to_vec())
+            .map_err(|_| FileStorageError::FailedToParseFingerprint)?;
+        let file_trie = RocksDbFileDataTrie::<T>::from_existing(self.storage.clone(), &mut root);
+
+        file_trie.get_chunk(chunk_id)
+    }
+
+    fn write_chunk(
+        &mut self,
+        key: &HasherOutT<T>,
+        chunk_id: &ChunkId,
+        data: &Chunk,
+    ) -> Result<FileStorageWriteOutcome, FileStorageWriteError> {
+        let raw_metadata = self
+            .storage
+            .read()
+            .expect("Failed to acquire read lock")
+            .read(*key)
+            .map_err(|_| FileStorageWriteError::FailedToReadStorage)?
+            .expect("Failed to find File Metadata");
+
+        let metadata: FileMetadata = serde_json::from_slice(&raw_metadata)
+            .map_err(|_| FileStorageWriteError::FailedToParseFileMetadata)?;
+        let raw_root = metadata.fingerprint.as_ref();
+        let mut root = convert_raw_bytes_to_hasher_out::<T>(raw_root.to_vec())
+            .map_err(|_| FileStorageWriteError::FailedToParseFingerprint)?;
+        let mut file_trie =
+            RocksDbFileDataTrie::<T>::from_existing(self.storage.clone(), &mut root);
+
+        file_trie
+            .write_chunk(chunk_id, data)
+            .map_err(|_| FileStorageWriteError::FailedToInsertFileChunk)?;
+
+        Ok(FileStorageWriteOutcome::FileComplete)
+    }
+
+    fn insert_file(
+        &mut self,
+        key: HasherOutT<T>,
+        metadata: FileMetadata,
+    ) -> Result<(), FileStorageError> {
+        let mut transaction = DBTransaction::new();
+        let serialized_metadata = serde_json::to_vec(&metadata)
+            .map_err(|_| FileStorageError::FailedToParseFileMetadata)?;
+
+        transaction.put(0, key.as_ref(), &serialized_metadata);
+
+        self.storage
+            .write()
+            .expect("Failed to acquire write lock")
+            .write(transaction)
+            .map_err(|_| FileStorageError::FailedToWriteToStorage)?;
+
+        Ok(())
+    }
+
+    fn insert_file_with_data(
```

Comment:
This function should not be doing any of the work. The file manager at this point should be a wrapper around file trie. File trie should be the one to insert a new trie into storage. Hence why I say we should probably remove storage from file manager, since there is no longer a distinct line of responsibility between both structs.

Reply:
You are right, I'm gonna remove the storage from File Storage in the next PR and give all the responsibility to the File Trie.